cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,292 @@
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            subscription_id = extract_sub_from_resourceid(azure["resourceId"])
+            logger.warning(
+                "extract_rapid7_configurations_subscriptionid0: %s (%s)",
+                subscription_id,
+                azure["resourceId"],
+            )
+
+    logger.warning("extract_rapid7_configurations_subscriptionid: %s", subscription_id)
+    return subscription_id
+
+
+def extract_rapid7_configurations_resourcegroup(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure resource group
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    rg = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            rg = extract_rg_from_resourceid(azure["resourceId"])
+
+    logger.warning("extract_rapid7_configurations_resourcegroup: %s", rg)
+    return rg
+
+
+def extract_rapid7_configurations_resourceid(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure resource id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    resource_id = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            resource_id = azure["resourceId"]
+
+    logger.warning("extract_rapid7_configurations_resourceid: %s", resource_id)
+    return resource_id
+
+
+def extract_rapid7_configurations_instanceid(configurations: Any[str, list]) -> str:
+    """
+    Extract from json, specific keyvalue pair: azure instance id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return ""
+
+    instance_id = ""
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            instance_id = azure["instanceId"]
+
+    logger.warning("extract_rapid7_configurations_instanceid: %s", instance_id)
+    return instance_id
+
+
+def extract_rapid7_history_lastseen(history: list) -> str:
```

Comment:
You can replace this with something like 
`min(entry['date'] for entry in history)`